### PR TITLE
Update RealEstateImporter.php

### PR DESCRIPTION
**Pull request by falcesch**
dynamic return child nodes

### CODE REVIEW

**Comment by zoglo** [comment]
Can you properly explain what this PR does?

**Comment by falcesch** [comment]
to select this
image
with **user_defined_simplefield@[kontakt_id]**

File: src/Resources/contao/classes/RealEstateImporter.php [hunk 1]
```diff
@@ -1150,38 +1152,26 @@ public function getFieldData($field, $group)
...
+                            foreach ($xmlNode->children() as $c) {
+                                if ($index === $attr_field) {
+                                    $results[$i] = $c->getName();
+                                }
+                                $index++;
                             }
-                            $index++;
+                        } else {
+                            // Returns the value of an multiple XML element.
+                            if(current($attributes)['feldname'] == $attr_field)
```
Comment (zoglo) [inline src/Resources/contao/classes/RealEstateImporter.php]: This may lead to an error if it doesn`t exist

**Review by zoglo — COMMENTED** [review]
- See below

**Review by zoglo — DISMISSED** [review]
I can not merge this as it is as it'll be a PITA to test this PR.
You also didn't add a proper description

File: src/Resources/contao/classes/RealEstateImporter.php (hunk 1 above)
Comment (falcesch) [reply src/Resources/contao/classes/RealEstateImporter.php]: then use this
if(array_key_first(current($attributes)) == $attr_field)

Comment (zoglo) [reply src/Resources/contao/classes/RealEstateImporter.php]: @falcesch
It is not my responsibility to address the issues you're trying to introduce with this PR.

**Comment by zoglo** [comment]
@falcesch 
I'm going to close this PR for the following reasons:

* The changes proposed in this PR affect a critical part of the software, and modifying the import could potentially cause significant issues. Merging this could lead to additional work on our side, as many clients depend on this software.

* The PR lacks a proper and clear description of its purpose and functionality

* It appears to be untested and lacks the necessary quality and testing. There also doesn't seem to be a willingness to properly collaborate and cooperate on this PR as seen in https://github.com/contao-estatemanager/core/pull/202#discussion_r1735930305

In my opinion, a better solution would be to use a hook, allowing you to implement your own logic.

If you require Pull requests like this to be implemented and would want us to implement them for you, please feel free to consult us at info@oveleon.de.

**Pull request closed without merging**